pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -131,8 +131,13 @@ public static StringBuilder escapeIdentifier(@Nullable StringBuilder sbuf, Strin
    * @param value value to append
    */
   private static void doAppendEscapedIdentifier(Appendable sbuf, String value) throws SQLException {
+
+    boolean alreadyQuoted = value.startsWith("\"") && value.endsWith("\"");
+
     try {
-      sbuf.append('"');
+      if ( !alreadyQuoted ) {
```

Comment:
Hm, I thought it should be more like 

```java
if (alreadyQuoted) {
  return;
}
```
Otherwise you might quote already quoted names because the for loop is still executed. This would result in badly quoted values that will fail executing, e.g. `"column"` as input will become `""column""` because of the for-loop.

Reply:
Well we need deal with internally quoted names

Reply:
Well ok, but now valid input will become broken escaped output, e.g. `"column"` will become `""column""` which always fails. Would make more sense if the for-loop would simply copy the first and last quote if `alreadyQuoted` is `true` without quoting it again. That way `"column"` stays `"column"` and `"column"name"` becomes `"column""name"`